test/powershell/Modules/Microsoft.PowerShell.Management/PSDrive.Tests.ps1

```
@@ -1,45 +1,137 @@
-Describe "Validate basic PSDrive Cmdlets" -Tags "CI" {
+Describe "Basic Alias Provider Tests" -Tags "CI" {
+    Context "Validate basic PSDrive Cmdlets" {
+        BeforeAll {
+            #just use same location as TestDrive for simplicity
+            $psDriveRoot = "TestDrive:"
+            $psDriveName = "PsTestDriveName"
+        }
+
+        BeforeEach {
+            New-PSDrive -Name $psDriveName -PSProvider FileSystem -Root $psDriveRoot > $null
+        }
+
+        AfterEach {
+            Remove-PSDrive -Name $psDriveName -ErrorAction SilentlyContinue
+        }
+
+        It "Create a new PSDrive" {
+            $newDrive = New-PSDrive -Name "NewDifferentPSDrive" -PSProvider FileSystem -Root $psDriveRoot
```

This should be safe to put within the try. The finally silently continues.